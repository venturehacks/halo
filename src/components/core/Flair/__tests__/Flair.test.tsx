import { render } from 'enzyme';
import React from 'react';

import { Flair } from '../index';

describe('Flair', () => {
  test('smoke', () => {
    const component = render(<Flair>content</Flair>);
    expect(component.text()).toEqual('content');
  });

  describe('snapshot', () => {
    test('default', () => {
      const component = render(<Flair>blue default</Flair>);
      expect(component).toMatchSnapshot();
    });

    describe('color', () => {
      test('orange', () => {
        const component = render(<Flair color="orange">orange</Flair>);
        expect(component).toMatchSnapshot();
      });

      test('red', () => {
        const component = render(<Flair color="red">red</Flair>);
        expect(component).toMatchSnapshot();
      });

      test('green', () => {
        const component = render(<Flair color="green">green</Flair>);
        expect(component).toMatchSnapshot();
      });
    });

    describe('size', () => {
      test('xs', () => {
        const component = render(<Flair size="xs">xs size</Flair>);
        expect(component).toMatchSnapshot();
      });

      test('sm', () => {
        const component = render(<Flair size="sm">sm size</Flair>);
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(<Flair size="md">md size</Flair>);
        expect(component).toMatchSnapshot();
      });
    });
  });
});
