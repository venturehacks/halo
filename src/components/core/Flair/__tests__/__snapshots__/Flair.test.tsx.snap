// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flair snapshot color green 1`] = `
<span
  class="component green sm"
>
  green
</span>
`;

exports[`Flair snapshot color orange 1`] = `
<span
  class="component orange sm"
>
  orange
</span>
`;

exports[`Flair snapshot color red 1`] = `
<span
  class="component red sm"
>
  red
</span>
`;

exports[`Flair snapshot default 1`] = `
<span
  class="component blue sm"
>
  blue default
</span>
`;

exports[`Flair snapshot size md 1`] = `
<span
  class="component blue md"
>
  md size
</span>
`;

exports[`Flair snapshot size sm 1`] = `
<span
  class="component blue sm"
>
  sm size
</span>
`;

exports[`Flair snapshot size xs 1`] = `
<span
  class="component blue xs"
>
  xs size
</span>
`;
